src/main/java/org/rumbledb/items/ObjectItem.java

```diff
@@ -116,13 +118,9 @@ public Item getItemByKey(String s) {
 
     @Override
     public void putItemByKey(String s, Item value) {
-        this.values.replaceAll(item -> {
-            if (this.values.indexOf(item) == this.keys.indexOf(s)) {
-                return value;
-            } else {
-                return item;
-            }
-        });
+        this.keys.add(s);
+        this.values.add(value);
+        checkForDuplicateKeys(this.keys, new ExceptionMetadata(0, 0));
```

Comment:
Is ExceptionMetadata for line 0:0 intentional here?

Reply:
yes, because this is outside of any JSONiq query.